frontend/src/app/supportAdmin/UnarchivePatients/UnarchivePatient.tsx

```diff
@@ -0,0 +1,425 @@
+        Filter by organization and testing facility to display archived
+        patients.
+      </div>
+    );
+  }
+};
+
+const displayNoFacilitiesMsg = (
+  selectedOrgId: string,
+  facilities: UnarchivePatientFacility[]
+) => {
+  if (selectedOrgId && facilities?.length === 0) {
+    return (
+      <div className="width-full margin-top-2">
+        This organization has no facilities. Select a different organization.
+      </div>
+    );
+  }
+};
+
+const getOrgExternalIdWithInternalId = (
+  orgs: UnarchivePatientOrganization[],
+  internalId: string
+) => {
+  let org = orgs.find((org) => org.internalId === internalId);
+  return org?.externalId;
+};
+const UnarchivePatient = ({
+  organizations,
+  currentPage,
+  selectedFacilityId,
+  setSelectedFacilityId,
+  selectedOrgId,
+  setSelectedOrgId,
+  archivedPatientsCount,
+  setArchivedPatientsCount,
+  facilities,
+  setFacilities,
+  archivedPatients,
+  setArchivedPatients,
+}: UnarchivePatientProps) => {
+  const pageTitle = `Unarchive ${PATIENT_TERM}`;
+  useDocumentTitle(pageTitle);
+  const unarchivePatientPageUrl = "/admin/unarchive-patient";
+  const entriesPerPage = 20;
+
+  const [isLoadingCount, setIsLoadingCount] = useState(false);
+  const [isLoadingPatients, setIsLoadingPatients] = useState(false);
+
+  const [activeFacility] = useSelectedFacility();
+  const navigate = useNavigate();
+
+  const [queryGetOrgWithFacilities] = useGetOrganizationWithFacilitiesLazyQuery(
+    {
+      fetchPolicy: "no-cache",
+    }
+  );
+  const [queryGetPatientsByFacilityWithOrg] =
+    useGetPatientsByFacilityWithOrgLazyQuery({
+      fetchPolicy: "no-cache",
+    });
+
+  const [queryGetPatientsCountByFacilityWithOrg] =
+    useGetPatientsCountByFacilityWithOrgLazyQuery({
+      fetchPolicy: "no-cache",
+    });
+
+  const {
+    getValues,
+    handleSubmit,
+    register,
+    watch,
+    formState: { errors },
+  } = useForm({});
+
+  const formCurrentValues = watch();
+
+  useEffect(() => {
+    fetchAndSetArchivedPatients(
+      selectedOrgId,
+      selectedFacilityId,
+      organizations
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentPage]);
+
+  const onOrgChange = async (selectedOrgInternalId: string) => {
+    setFacilities([]);
+    setArchivedPatients(undefined);
+    setArchivedPatientsCount(undefined);
+    if (selectedOrgInternalId) {
+      setSelectedOrgId(selectedOrgInternalId);
+      let res = await queryGetOrgWithFacilities({
+        variables: { id: selectedOrgInternalId },
+      });
+      let facilities = res.data?.organization?.facilities
+        ? res.data?.organization?.facilities
+        : [];
+      if (facilities !== undefined && facilities.length > 0) {
+        facilities.sort((a, b) => (a.name > b.name ? 1 : -1));
+      }
+      setFacilities(facilities);
+    }
+  };
+
+  const onFacilityChange = (facilityId: string) => {
+    if (facilityId === "") {
+      setArchivedPatientsCount(undefined);
```

Comment:
⚠️ to address when someone clicks "Select a testing facility" option from the dropdown

Reply:
no longer applicable